### PR TITLE
修改COPY-FROM No. 18 static

### PR DESCRIPTION
**Pull request by enkilee**
修改COPY-FROM No. 18 static
PADDLEPADDLE_PR=54842
* https://github.com/PaddlePaddle/docs/issues/5957#issue
* https://github.com/PaddlePaddle/Paddle/pull/54842

### CODE REVIEW

File: docs/api/paddle/static/Program_cn.rst [hunk 1]
```diff
@@ -12,7 +12,7 @@ Program
 Program 是 Paddle 对于计算图的一种静态描述，使用 Program 的构造函数可以创建一个 Program。Program 中包括至少一个 :ref:`api_guide_Block`，当 :ref:`api_guide_Block` 中存在条件选择的控制流 OP（例如 :ref:`cn_api_fluid_layers_While` 等）时，该 Program 将会含有嵌套着的 :ref:`api_guide_Block` 即控制流外部的 :ref:`api_guide_Block` 将包含着控制流内部的 :ref:`api_guide_Block`，而嵌套的 :ref:`api_guide_Block` 的元素访问控制将由具体的控制流 OP 来决定。关于 Program 具体的结构和包含的类型请参阅 `framework.proto <https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/framework.proto>`_
 。
 
-一个 Program 的集合通常包含初始化程序（startup_program）与主程序(main_program)，初始化程序是一个包含一些初始化工作的 Program，主程序将会包含用来训练的网络结构和变量，在使用同一个 :ref:`api_guide_executor` 执行时他们会共享初始化工作的结果，例如初始化的参数。一个 Program 的集合可以被用来测试或者训练，被用来训练时，``Paddle`` 将会利用所有用户使用的 OP 和变量来搭建一个训练网络，被用来测试时，可以通过调用 Program 相关的接口例如：`clone` 剪去一些与测试无关的 OP 和变量，比如反向传播的 OP 和变量。
+一个 Program 的集合通常包含初始化程序(startup_program)与主程序(main_program)，初始化程序是一个包含一些初始化工作的 Program，主程序将会包含用来训练的网络结构和变量，在使用同一个 :ref:`api_guide_executor` 执行时他们会共享初始化工作的结果，例如初始化的参数。一个 Program 的集合可以被用来测试或者训练，被用来训练时，``Paddle`` 将会使用所有用户的 OP 和变量来搭建一个训练网络，被用来测试时，可以通过调用 Program 相关的接口例如：``clone` 剪去一些与测试无关的 OP 和变量，比如反向传播的 OP 和变量。
```
Comment (SigureMo) [inline docs/api/paddle/static/Program_cn.rst]: - 中文里用中文全角括号
- `将会利用所有用户使用的` 和 `将会使用所有用户的` 表达的意思不一样吧
- \`\`clone\` -> \`\`clone\`\`

File: docs/api/paddle/static/Program_cn.rst [hunk 2]
```diff
@@ -82,29 +54,16 @@ clone(for_test=False)
     2. 此 API 当 ``for_test=True`` 时将会裁剪部分 OP 和变量。为防止错误的裁剪，推荐在 :ref:`cn_api_fluid_backward_append_backward` 和执行优化器之前使用；``clone(for_test=True)`` 。
 
 
-当 ``for_test=True`` 时创建一个新的、仅包含当前 Program 前向内容的 Program。否则创建一个新的，和当前 Program 完全相同的 Program
+当 ``for_test=True`` 时创建一个新的、仅包含当前 Program 前向内容的 Program。否则创建一个新的和当前 Program 完全相同的 Program 。
```
Comment (SigureMo) [inline docs/api/paddle/static/Program_cn.rst]: 删去 Program 和 `。` 之间的空格

File: docs/api/paddle/static/Program_cn.rst [hunk 3]
```diff
@@ -477,7 +217,7 @@ state_dict(mode='all', scope=None)
 
 **参数**
 
-    - **mode** (str，可选) - 获取何种持久性变量。目前支持以下选项：(1) 'opt'：获得优化器的持久性变量放在 dict 结构中；(2) 'param'：获得组网中的持久性变量放在 dict 结构中，不包含优化器中的持久性变量；(3) 'all'：获得组网和优化器中的持久性变量放在 dict 结构中；默认值为'all'。
+    - **mode** (str，可选) - 获取何种持久性变量。目前支持以下选项：(1) ``opt`` ：获得优化器的持久性变量放在 ``dict`` 结构中；(2) ``param``：获得组网中的持久性变量放在 ``dict`` 结构中，不包含优化器中的持久性变量；(3) ``all`` ：获得组网和优化器中的持久性变量放在 dict 结构中；默认值为 ``all`` 。
```
Comment (SigureMo) [inline docs/api/paddle/static/Program_cn.rst]:
```suggestion
    - **mode** (str，可选) - 获取何种持久性变量。目前支持以下选项：(1) ``opt``：获得优化器的持久性变量放在 ``dict`` 结构中；(2) ``param``：获得组网中的持久性变量放在 ``dict`` 结构中，不包含优化器中的持久性变量；(3) ``all``：获得组网和优化器中的持久性变量放在 dict 结构中；默认值为 ``all``。
```

File: docs/api/paddle/static/nn/cond_cn.rst [hunk 4]
```diff
@@ -18,60 +18,22 @@ PaddlePaddle 里 Tensor 的嵌套结构是指一个 Tensor，或者 Tensor 的
...
+       COPY-FROM: paddle.static.nn.cond:code-example1
 
        不管 ``a < b`` 是否成立，``c = a * b`` 都会被组网且运行，``a + c`` 和 ``b * b`` 都会参与组网，只是组网后运行时只会运行条件对应的分支。
 
 参数
 :::::::::
     - **pred** (Tensor) - 一个元素个数为 1 的布尔型（boolean）的 Tensor （ 0-D Tensor 或者形状为 [1] ），该布尔值决定要返回 ``true_fn`` 还是 ``false_fn`` 的运行结果。
-    - **true_fn** (callable) - 一个当 ``pred`` 是 ``True`` 时被调用的 callable，默认值：``None`` 。
-    - **false_fn** (callable) - 一个当 ``pred`` 是 ``False`` 时被调用的 callable，默认值：``None`` 。
+    - **true_fn** (callable) - 一个当 ``pred`` 是 ``True`` 时被调用的 callable，默认值： ``None`` 。
```
Comment (SigureMo) [inline docs/api/paddle/static/nn/cond_cn.rst]:
```suggestion
    - **true_fn** (callable) - 一个当 ``pred`` 是 ``True`` 时被调用的 callable，默认值：``None``。
```

File: docs/api/paddle/static/nn/cond_cn.rst [hunk 5]
```diff
@@ -18,60 +18,22 @@ PaddlePaddle 里 Tensor 的嵌套结构是指一个 Tensor，或者 Tensor 的
...
+    - **false_fn** (callable) - 一个当 ``pred`` 是 ``False`` 时被调用的 callable，默认值： ``None`` 。
```
Comment (SigureMo) [inline docs/api/paddle/static/nn/cond_cn.rst]:
```suggestion
    - **false_fn** (callable) - 一个当 ``pred`` 是 ``False`` 时被调用的 callable，默认值：``None``。
```

File: docs/api/paddle/static/nn/cond_cn.rst [hunk 6]
```diff
@@ -18,60 +18,22 @@ PaddlePaddle 里 Tensor 的嵌套结构是指一个 Tensor，或者 Tensor 的
...
     - **name** (str，可选) - 具体用法请参见 :ref:`api_guide_Name`，一般无需设置，默认值为 None。
+    - **return_names** (sequence of string，可选) - 通常情况下，用户不必设置此参数。字符串表示返回的变量的名称。序列的结构必须与 ``true_fn`` 和 ``false_fn`` 的返回值相同，默认值： ``None`` 。
```
Comment (SigureMo) [inline docs/api/paddle/static/nn/cond_cn.rst]:
```suggestion
    - **return_names** (sequence of string，可选) - 通常情况下，用户不必设置此参数。字符串表示返回的变量的名称。序列的结构必须与 ``true_fn`` 和 ``false_fn`` 的返回值相同，默认值：``None``。
```

File: docs/api/paddle/static/nn/fc_cn.rst [hunk 7]
```diff
@@ -65,6 +65,7 @@ fc
     out.shape = (1, 2)
 
 
+
```
Comment (SigureMo) [inline docs/api/paddle/static/nn/fc_cn.rst]: 为什么要加这个空行？

File: docs/api/paddle/static/Program_cn.rst [hunk 8]
```diff
@@ -50,7 +35,7 @@ to_string(throw_on_error, with_details=False)
 **参数**
 
  - **throw_on_error** (bool) - 是否在没有设置必需字段时抛出异常。
- - **with_details** (bool) - 值为 true 时，打印更多关于变量和参数的信息，如 trainable, optimize_attr 等。
+ - **with_details** (bool) - 值为 true 时，打印更多关于变量和参数的信息，如 ``trainable``, ``optimize_attr``等。
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Program_cn.rst]:
```suggestion
 - **with_details** (bool) - 值为 true 时，打印更多关于变量和参数的信息，如 ``trainable``, ``optimize_attr`` 等。
```
image

File: docs/api/paddle/static/Program_cn.rst [hunk 9]
```diff
@@ -82,29 +54,16 @@ clone(for_test=False)
...
-        paddle.enable_static()
-
-        img = static.data(name='image', shape=[None, 784])
-        pred = static.nn.fc(x=img, size=10, activation='relu')
-        loss = paddle.mean(pred)
-        # Here we use clone before Momentum
-        test_program = static.default_main_program().clone(for_test=True)
-        optimizer = paddle.optimizer.Momentum(learning_rate=0.01, momentum=0.9)
-        optimizer.minimize(loss)
+COPY-FROM: paddle.static.Program:code-example1
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Program_cn.rst]: 应该是clone方法下的示例代码叭
```suggestion
COPY-FROM: paddle.static.Program.clone:code-example1
```
image

File: docs/api/paddle/static/Program_cn.rst [hunk 10]
```diff
@@ -120,116 +79,15 @@ Program，当 ``for_test=True`` 时返回一个新的、仅包含当前 Program
...
-        for name, value in sorted(prog.block(0).vars.items()):
-            print(value)
-        for op in prog.block(0).ops:
-            print("op type is {}".format(op.type))
-            print("op inputs are {}".format(op.input_arg_names))
-            print("op outputs are {}".format(op.output_arg_names))
-            for key, value in sorted(op.all_attrs().items()):
-                if key not in ['op_callstack', 'op_role_var']:
-                    print(" [ attrs: {}:   {} ]".format(key, value))
+COPY-FROM: paddle.static.Program:code-example2
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Program_cn.rst]:
```suggestion
COPY-FROM: paddle.static.Program.clone:code-example2
```

File: docs/api/paddle/static/Program_cn.rst [hunk 11]
```diff
@@ -120,116 +79,15 @@ Program，当 ``for_test=True`` 时返回一个新的、仅包含当前 Program
...
-    # In Paddle we will share weights by using the same Tensor name. In train and test program
-    # all parameters will have the same name and this can make train and test program sharing parameters,
-    # that's why we need to use startup program of train. And for startup program of test, it has nothing,
-    # since it is a new program.
-
-    with static.program_guard(train_program, startup_program):
-        with utils.unique_name.guard():
-            sgd = paddle.optimizer.SGD(learning_rate=1e-3)
-            sgd.minimize(avg_loss)
+COPY-FROM: paddle.static.Program:code-example3
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Program_cn.rst]:
```suggestion
COPY-FROM: paddle.static.Program.clone:code-example3
```

File: docs/api/paddle/static/Program_cn.rst [hunk 12]
```diff
@@ -120,116 +79,15 @@ Program，当 ``for_test=True`` 时返回一个新的、仅包含当前 Program
...
-        with utils.unique_name.guard():
-            avg_loss = network()
-            sgd = paddle.optimizer.SGD(learning_rate=1e-3)
-            sgd.minimize(avg_loss)
-    # the test startup program is not used.
-    with static.program_guard(test_program_2, startup_program_2):
-        with utils.unique_name.guard():
-            avg_loss = network()
-    print_prog(test_program_2)
+COPY-FROM: paddle.static.Program:code-example4
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Program_cn.rst]:
```suggestion
COPY-FROM: paddle.static.Program.clone:code-example4
```

File: docs/api/paddle/static/Variable_cn.rst [hunk 13]
```diff
@@ -21,17 +21,7 @@ Variable
...
-      import paddle
-
-      paddle.enable_static()
-
-      cur_program = paddle.static.Program()
-      cur_block = cur_program.current_block()
-      new_variable = cur_block.create_var(name="X",
-                                          shape=[-1, 23, 48],
-                                          dtype='float32')
+COPY-FROM: paddle.static.Variable:code-example1
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Variable_cn.rst]: - 没有成功引入代码，看了下variable继承了VariableMetaClass，修改成
 `COPY-FROM: paddle.static.VariableMetaClass.Variable:code-example1`
image

- 英文中有两种情况，分别有代码块，这儿也同步引用两种叭
image

File: docs/api/paddle/static/Variable_cn.rst [hunk 14]
```diff
@@ -150,33 +105,8 @@ Tensor， :ref:`api_guide_Variable` 的值。
...
-      for var in prog.list_vars():
-          if var.persistable:
-              t = var.get_value()
-              paddle.save(t, path+var.name+'.pdtensor')
-
-      for var in prog.list_vars():
-          if var.persistable:
-              t_load = paddle.load(path+var.name+'.pdtensor')
-              var.set_value(t_load)
+**代码示例**
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/Variable_cn.rst]: image
代码示例重复了

```suggestion
```

File: docs/api/paddle/static/Variable_cn.rst
Comment (sunzhongkai588) [inline docs/api/paddle/static/Variable_cn.rst:1]: `ndimension(self)` 和 `dim(self)` 这两个方法的代码示例仍没有使用copy-from，请辛苦修改～

File: docs/api/paddle/static/nn/cond_cn.rst [hunk 15]
```diff
@@ -4,74 +4,36 @@ cond
...
-
-    x = paddle.full(shape=[1], dtype='float32', fill_value=0.1)
-    y = paddle.full(shape=[1], dtype='float32', fill_value=0.23)
-    pred = paddle.less_than(x=x, y=y, name=None)
-    ret = paddle.static.nn.cond(pred, true_func, false_func)
-    # ret 是包含两个 tensors 的元组
-    # ret[0] = [[1 1]]
-    # ret[1] = [[ True  True  True]
-    #           [ True  True  True]]
+COPY-FROM: paddle.static.nn.cond
```
Comment (sunzhongkai588) [inline docs/api/paddle/static/nn/cond_cn.rst]: 应在英文PR里该出增加name标识，则该处也同步修改为
```suggestion
COPY-FROM: paddle.static.nn.cond:code-example2
```

File: docs/api/paddle/static/program_guard_cn.rst
Comment (sunzhongkai588) [inline docs/api/paddle/static/program_guard_cn.rst:1]: @SigureMo 似乎这块儿的copy-from填写正确，但无法显示代码，帮忙查看一下
image

Comment (SigureMo) [reply docs/api/paddle/static/program_guard_cn.rst:1]: > `PADDLEPADDLE_PR=https://github.com/PaddlePaddle/Paddle/pull/54842`

PR 描述里填写的就不对吧

File: docs/api/paddle/static/Variable_cn.rst
Comment (enkilee) [reply docs/api/paddle/static/Variable_cn.rst:1]: 这2方法，在英文中没有啊，直接删吗？

File: docs/api/paddle/static/Program_cn.rst (hunk 9 above)
Comment (enkilee) [reply docs/api/paddle/static/Program_cn.rst]: OK

**Comment by enkilee** [comment]
@SigureMo 佬，麻烦有空试下 dim 和 ndimension , 直接改怎么都通不过，改回lambda就没问题。

**Comment by SigureMo** [comment]
中英文 PR 要互链

File: docs/api/paddle/static/Program_cn.rst [hunk 16]
```diff
@@ -50,7 +35,7 @@ to_string(throw_on_error, with_details=False)
 **参数**
 
  - **throw_on_error** (bool) - 是否在没有设置必需字段时抛出异常。
- - **with_details** (bool) - 值为 true 时，打印更多关于变量和参数的信息，如 trainable, optimize_attr 等。
+ - **with_details** (bool) - 值为 true 时，打印更多关于变量和参数的信息，如 ``trainable``, ``optimize_attr`` 等。
```
Comment (SigureMo) [inline docs/api/paddle/static/Program_cn.rst]:
```suggestion
 - **with_details** (bool) - 值为 true 时，打印更多关于变量和参数的信息，如 ``trainable``，``optimize_attr`` 等。
```

中文里用中文标点

Comment (enkilee) [reply docs/api/paddle/static/Program_cn.rst]: 收到

**Review by sunzhongkai588 — APPROVED** [review]
LGTM
